protobuf-bean/src/main/java/io/apicurio/registry/examples/simple/protobuf/ProtobufBeanExample.java

```diff
@@ -0,0 +1,200 @@
+                });
+            }
+        } finally {
+            consumer.close();
+        }
+
+        RegistryClient client = RegistryClientFactory.create(REGISTRY_URL);
+        System.out.println("The artifact created in Apicurio Registry is: ");
+        //because the default ArtifactResolverStrategy is TopicIdStrategy the artifactId is in the form of topicName-value
+        System.out.println(IoUtil.toString(client.getArtifactVersion("default", topicName + "-value", "1")));
+        System.out.println();
+        System.out.println("Done (success).");
+    }
+
+    /**
+     * Creates the Kafka producer.
+     */
+    private static Producer<Object, AddressBook> createKafkaProducer() {
+        Properties props = new Properties();
+
+        // Configure kafka settings
+        props.putIfAbsent(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, SERVERS);
+        props.putIfAbsent(ProducerConfig.CLIENT_ID_CONFIG, "Producer-" + TOPIC_NAME);
+        props.putIfAbsent(ProducerConfig.ACKS_CONFIG, "all");
+        props.putIfAbsent(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName());
+        // Use the Apicurio Registry provided Kafka Serializer for Protobuf
+        props.putIfAbsent(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, ProtobufKafkaSerializer.class.getName());
+
+        // Configure Service Registry location
+        props.putIfAbsent(SerdeConfig.REGISTRY_URL, REGISTRY_URL);
+        props.putIfAbsent(SerdeConfig.EXPLICIT_ARTIFACT_GROUP_ID, "default");
+
+        // Register the artifact if not found in the registry.
+        props.putIfAbsent(SerdeConfig.AUTO_REGISTER_ARTIFACT, Boolean.TRUE);
+
+        // Create the Kafka producer
+        Producer<Object, AddressBook> producer = new KafkaProducer<>(props);
+        return producer;
+    }
+
+    /**
+     * Creates the Kafka consumer.
+     */
+    private static KafkaConsumer<Long, AddressBook> createKafkaConsumer() {
+        Properties props = new Properties();
+
+        // Configure Kafka
+        props.putIfAbsent(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, SERVERS);
+        props.putIfAbsent(ConsumerConfig.GROUP_ID_CONFIG, "Consumer-" + TOPIC_NAME);
+        props.putIfAbsent(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "true");
+        props.putIfAbsent(ConsumerConfig.AUTO_COMMIT_INTERVAL_MS_CONFIG, "1000");
+        props.putIfAbsent(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+        props.putIfAbsent(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class.getName());
+        // Use the Apicurio Registry provided Kafka Deserializer for Protobuf
+        props.putIfAbsent(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, ProtobufKafkaDeserializer.class.getName());
+
+        // Configure Service Registry location
+        props.putIfAbsent(SerdeConfig.REGISTRY_URL, REGISTRY_URL);
+        props.putIfAbsent(SerdeConfig.EXPLICIT_ARTIFACT_GROUP_ID, "default");
```

Comment:
Is this needed?  I assumed the deserializer wouldn't need this.

Reply:
no, it's not need, I'll remove it